keras_lr_finder/lr_finder.py

```diff
@@ -62,6 +62,36 @@ def find(self, x_train, y_train, start_lr, end_lr, batch_size=64, epochs=1):
         # Restore the original learning rate
         K.set_value(self.model.optimizer.lr, original_lr)
 
+    def find_generator(self, generator, start_lr, end_lr, epochs=1, num_batches=None):
+            if num_batches is None:
+                num_batches = len(generator)
```

Comment:
Sorry, but there are a few of issues with this.
1) len works with sequences or collections, but might not work with arbitrary iterables. AFAIK, the class must implement __len__ method in order for len to work.
2) If the generator is infinite then this line will never complete
3) Generator might not be rerunnable. This line might exhaust the generator, and then fit_generator will fail.

I would suggest making num_batches a required parameter.

Reply:
Don't be sorry! In general I agree with you, but in this case I am using `len(generator)` to be consistent with Keras interface [https://keras.io/models/sequential/#fit_generator](url). To be even more consistent, I have changed the name from `num_batches` to `steps_per_epoch`, and I have added a check to catch the case of a generator without `__len__()`.